smartsim/experiment.py

```diff
@@ -846,3 +880,27 @@ def _launch_summary(self, manifest: Manifest) -> None:
 
     def __str__(self) -> str:
         return self.name
+
+    def dbs_in_use(self) -> set():
+        return set(self.my_list)
+
+    def dbs_in_use_after_start(self) -> set():
+        return set(self.my_list_after_start)
+
+    def dbs_list(self, db_identifier):
+        # check that using list matters
+        self.my_list.append(db_identifier)
+        return self.my_list
+
+    def dbs_list_after_start(self, db_identifier):
+        # either use self or dont use self..
+        self.my_list_after_start.append(db_identifier)
+        return self.my_list
+
+    def check_db_identifier_unique(self, flag):
+        """check that db_identifiers are unique
+        Doing this a weird way?"""
+
+        if flag == "before":
+            return len(set(self.my_list)) == len(self.my_list)
+        return len(set(self.my_list_after_start)) == len(self.my_list_after_start)
```

Comment:
I think it would be better to check for set membership?

Reply:
Easiest/clearest way, if you do this in the function-to-be-renamed `dbs_list` above is to check if `db_identifier` is already in the list